packages/next/src/build/webpack-config.ts

```diff
@@ -2024,6 +2041,66 @@ export default async function getBaseWebpackConfig(
             ]
           },
         },
+        // Alias server-only and client-only to proper exports based on bundling layers
+        {
+          issuerLayer: {
+            or: WEBPACK_LAYERS.GROUP.serverTarget,
+          },
+          resolve: {
+            // Error on client-only but allow server-only
+            alias: {
+              'server-only$': 'next/dist/compiled/server-only/empty',
+              'client-only$': 'next/dist/compiled/client-only/error',
+              'next/dist/compiled/server-only$':
+                'next/dist/compiled/server-only/empty',
+            },
+          },
+        },
+        {
+          issuerLayer: {
+            not: WEBPACK_LAYERS.GROUP.serverTarget,
+          },
+          resolve: {
+            // Error on server-only but allow client-only
+            alias: {
+              'server-only$': 'next/dist/compiled/server-only/index',
```

Comment:
should this be pointing to `error` based on the comment above?

Reply:
Yeah this exports files was bit different between client-only and server-only, sometimes confusing